samza-core/src/main/java/org/apache/samza/job/model/JobModel.java

```diff
@@ -39,34 +36,14 @@
  * </p>
  */
 public class JobModel {
-  private static final String EMPTY_STRING = "";
   private final Config config;
   private final Map<String, ContainerModel> containers;
-
-  private final LocalityManager localityManager;
-  private final Map<String, String> localityMappings;
```

Comment:
What do you think about leaving this in here (but make it fully immutable)? Would we be able to always read the localityMappings before creating a JobModel?
It seems like the localityMappings might be a conceptual fit in the JobModel. I kind of see it a bit similarly to the container-to-task mappings which are in the ContainerModel.
Other historical info: One thing that we considered when doing the Context API changes was to add JobModel to the JobContext, but that required putting JobModel in samza-api, and that didn't work due to LocalityManager still being there. However, now that LocalityManager is being removed, maybe we could move JobModel. It might be useful to provide access to the locality mappings in the Context. On the other hand, we could just leave it out for now and then add it later.